index.js

```diff
@@ -1,43 +1,58 @@
-'use strict';
+'use strict'
 
-import { Platform } from 'react-native';
-import StackTrace from 'stacktrace-js';
-import { Crashlytics } from 'react-native-fabric';
-var assign = require('lodash.assign');
+import StackTrace from 'stacktrace-js'
+import {initSourceMaps, getStackTrace} from 'react-native-source-maps'
+import { Crashlytics } from 'react-native-fabric'
+var assign = require('lodash.assign')
 
-function init() {
+function init (sourcemapName = null) {
   if (__DEV__) {
     // Don't send exceptions from __DEV__, it's way too noisy!
     // Live reloading and hot reloading in particular lead to tons of noise...
-    return;
+    return
   }
 
-  var originalHandler = global.ErrorUtils.getGlobalHandler();
-  function errorHandler(e, isFatal) {
-    StackTrace.fromError(e, {offline: true}).then((x) => {
-      Crashlytics.recordCustomExceptionName(e.message, e.message, x.map((row) => (assign({}, row, {
-        fileName: `${row.fileName}:${row.lineNumber || 0}:${row.columnNumber || 0}`,
-      }))))
-    });
+  var originalHandler = global.ErrorUtils.getGlobalHandler()
+  if (sourcemapName) {
+    initSourceMaps({sourceMapBundle: sourcemapName})
+  }
+  async function errorHandler (e, isFatal) {
+    let jsParsed = false
+    // map the JS stacktrace using the sourcemap file given on init
+    try {
+      let trace = null
+      if (!sourcemapName || !(trace = await getStackTrace(e))) {
+        // record the exception with the info that we have
+        let x = await StackTrace.fromError(e, {offline: true})
+        trace = x.map((row) => (assign({}, row, {
+          fileName: `${row.fileName}:${row.lineNumber || 0}:${row.columnNumber || 0}`
+        })))
+      }
+      if (trace) {
+        e.stack = trace
+        Crashlytics.recordCustomExceptionName(e.message, e.message, e.stack)
+        jsParsed = true
+      }
+    } catch (error) {
```

Comment:
What is this try/catch for? Seems like it would catch any errors in our reporting system...and silently drop them. This seems like a bad idea? Do you know what this is actually catching?

Reply:
The try/catch block was based on the Readme file of https://www.npmjs.com/package/react-native-source-maps. But yeah, I guess it should be only around the line for `await getStackTrace()`. Do you want me to change it? Because I'm currently on vacation and my computer was formatted yesterday and so, I will take a bit more time to look on this again.